src/rpc/request.cpp

```diff
@@ -134,8 +134,11 @@ static fs::path GetMainchainAuthCookieFile()
     if (gArgs.GetChainName() == "liquidv1") {
         cookie_file = ".cookie";
     }
-    std::string path = gArgs.GetArg("-mainchainrpccookiefile", cookie_file);
-    return AbsPathForConfigVal(fs::path(path));
+    std::string path = gArgs.GetArg("-mainchainrpccookiefile", "");
+    if (path == "") {
+        return fsbridge::AbsPathJoin(GetMainchainDefaultDataDir(), cookie_file);
+    }
+    return AbsPathForConfigVal(fs::path(cookie_file));
 }
```

Comment:
This doesn't work right if `-mainchainrpccookiefile` is specified, which is presumably why CI is failing.

Instead of this, I would suggest just replacing the line 

`cookie_file = ".cookie";`

Which sets the default, with 

`cookie_file = fsbridge::AbsPathJoin(GetMainchainDefaultDataDir(), ".cookie");`

So that the default is to use the cookie file in the right place, but the logic for handling user arguments with GetArg is unchanged.